### PR TITLE
Remove butter knife dependency.

### PR DESCRIPTION
**Pull request by mrkcsc**
This PR drops butter knife to help minimize issues with pro guard when consuming the library and pave the path for a complete `Kotlin` re-vamp.

### CODE REVIEW

File: app/src/main/java/com/lytefast/flexinput/sampleapp/MainFragment.java [hunk 1]
```diff
@@ -57,13 +52,15 @@ public void onCreate(Bundle savedInstanceState) {
   public View onCreateView(LayoutInflater inflater, ViewGroup container,
                            Bundle savedInstanceState) {
     View view = inflater.inflate(R.layout.fragment_message_main, container, false);
+
```
Comment (lytefast) [inline app/src/main/java/com/lytefast/flexinput/sampleapp/MainFragment.java]: nit: this double spacing doesn't fly in my codebase :P
remove top spacing only

File: flexinput/build.gradle [hunk 2]
```diff
@@ -38,9 +33,7 @@ dependencies {
     compile "com.android.support:support-v4:$supportLibraryVersion"
     compile "com.android.support.constraint:constraint-layout:$supportLibraryConstraintVersion"
 
-    compile "com.jakewharton:butterknife:8.4.0"
-    annotationProcessor "com.jakewharton:butterknife-compiler:8.4.0"
-
+    //noinspection GradleDynamicVersion
```
Comment (lytefast) [inline flexinput/build.gradle:36]: ? remove?

File: flexinput/src/main/java/com/lytefast/flexinput/fragment/FlexInputFragment.java [hunk 3]
```diff
@@ -152,7 +145,70 @@ public void run() {
...
+    attachmentClearButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        clearAttachments();
+      }
+    });
+    inputContainer = root.findViewById(R.id.main_input_container);
+    emojiContainer = root.findViewById(R.id.emoji_container);
+    attachmentPreviewList = root.findViewById(R.id.attachment_preview_list);
+    textEt = root.findViewById(R.id.text_input);
```
Comment (lytefast) [inline flexinput/src/main/java/com/lytefast/flexinput/fragment/FlexInputFragment.java]: nit: would prefer for this to be in a `bindTextInput(textEt)`

File: flexinput/src/main/java/com/lytefast/flexinput/fragment/FlexInputFragment.java [hunk 4]
```diff
@@ -152,7 +145,70 @@ public void run() {
...
+        updateSendBtnEnableState(editable);
+      }
+    });
+    textEt.setOnTouchListener(new View.OnTouchListener() {
+      @Override
+      public boolean onTouch(View view, MotionEvent motionEvent) {
+        return onTextInputTouch(motionEvent);
+      }
+    });
+    emojiBtn = root.findViewById(R.id.emoji_btn);
```
Comment (lytefast) [inline flexinput/src/main/java/com/lytefast/flexinput/fragment/FlexInputFragment.java]: nit would prefer for the buttons to be in a `bindButtons(root)`

**Review by lytefast — APPROVED** [review]

File: flexinput/build.gradle (hunk 2 above)
Comment (mrkcsc) [reply flexinput/build.gradle:36]: It's a warning otherwise but if you don't mind it I can remove.

Comment (lytefast) [reply flexinput/build.gradle:36]: I don't mind

**Pull request merged**